jetty-server/src/main/java/org/eclipse/jetty/server/AsyncAttributes.java

```diff
@@ -94,54 +83,15 @@ public void setAttribute(String key, Object value)
         switch (key)
         {
             case AsyncContext.ASYNC_REQUEST_URI:
-                _requestURI = (String)value;
-                break;
             case AsyncContext.ASYNC_CONTEXT_PATH:
-                _contextPath = (String)value;
-                break;
             case AsyncContext.ASYNC_SERVLET_PATH:
-                _servletPath = (String)value;
-                break;
             case AsyncContext.ASYNC_PATH_INFO:
-                _pathInfo = (String)value;
-                break;
             case AsyncContext.ASYNC_QUERY_STRING:
-                _queryString = (String)value;
-                break;
             case AsyncContext.ASYNC_MAPPING:
-                _httpServletMapping = (HttpServletMapping)value;
                 break;
             default:
                 super.setAttribute(key, value);
                 break;
         }
```

Comment:
Different behaviour here than ForwardAttribute and IncludeAttribute:  both of those do a remove if the value is null.

Reply:
Added comment and simplified forward/include.